### ISSUE TITLE
VS Code "Do to definition" for C/AL objects could use improvement

### ISSUE THREAD

**Issue by CanuckNav74** [Microsoft/AL#4258]
If I'm asked to develop a page extension for page I have little familiarity with, I was hoping that by right clicking on the relevant base object in one my my AL files and go to the definition for it, that it would give me all the useful info I need about that base object. It doesn't. 

example .al pseudocode:

```
pageextension 50001 "ATL Time Sheet" extends "Time Sheet"
{
...
}
```

In the sample above I would normally right-click on "Time Sheet" to go to the definition for the Time Sheet page object (page 950 in this case). When I do that, it works in theory (vs code spits out a .dal file for that C/AL object), but there is a lot of useful info missing that I believe should also appear. 

All that is rendered for the .dal file is the page layout, actions, and global variables.

There's a great deal of missing information that would be othwerise useful to see from within VS code (e.g. page properties such as SourceTable), external global functions, and especially event publications. All of those useful details are absent in the .dal file that VS Code displays. 

Currently this means if I want to really know all the "meat" in a C/AL object, I still need to spin up cside for my BC docker container to inspect it. I still need two environments to do my work (cside and vscode).

**Comment by CanuckNav74** [comment]
I'll test it out when I'm back in the office.

Having one launch.json config isn't a long term answer though.

I typically keep four configs in mine .

Two for Docker (one for recreate, one for synchronize)

To for the tenant's SaaS sandbox (one for recreate, one for synchronize)

**Comment by kalberes** [comment]
See #4203. For now go to definition will not retrieve a full base object in dal format if you have multiple configurations active in your launch.json.

**Comment by CanuckNav74** [comment]
thanks for the update @kalberes 

This presents a catch-22 problem if that's the case. 

* SaaS sandboxes are only debuggable by deploying an AL .app from VS code
* SaaS sandboxes currently do not support extension "upgrade" functionality
* to debug an AL extension upgrade, it is suggested that a local docker BC container is needed

Based on this, it seems that a launch.json for any given dev environment would need at least two launch configurations (one for the tenant, one for docker). Its unfortunate that the .dal design doesn't support this workflow. 

I suppose for now I'll have to add launch.json to my .gitignore and add the following files to the .vscode dir (and simply copy/paste the relevant launch contents into launch.json when its time to debug). Not ideal but it looks like it's the only workaround for now. 

* launch.sandbox.recreate.json
* launch.sandbox.synchronize.json
* launch.docker.recreate.json
* launch.docker.synchronize.json

**Comment by kalberes** [comment]
Yes, you are right. It is less then  sub-optimal.  I will mark the referenced one as a bug and we will fix it within the next releases.

**Comment by CanuckNav74** [comment]
@IvoMol I've repro'd your suggestion (a launch.json with only one launch config produces a detailed object.dal file with nothing missing.

**Comment by CanuckNav74** [comment]
excellent thank you @kalberes 

I agree - if a dialog/prompt appears (i.e. no different than F5 / F6 / download symbols) then I'd be fine with that.

**Comment by EmilDamsbo** [comment]
This issue has been addressed but it seems we didn't close out this issue. I'll close it out now.
